src/nvim/misc1.c

```diff
@@ -1067,3 +1067,32 @@ void add_time(char_u *buf, size_t buflen, time_t tt)
                  seconds);
   }
 }
+
+/// Fires a ModeChanged autocmd.
+void trigger_modechanged(void)
```

Comment:
`get_mode` is in `state.c`, so maybe you can put this in `state.c` as well.

Reply:
Moving can be done in a separate PR after the patches for moving code to separate files are done to ease porting.